src/config.rs

```diff
@@ -0,0 +1,96 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    fs::{self, File},
+    io::{ErrorKind, Write},
+    net::Ipv4Addr,
+    path::PathBuf,
+};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Config {
+    /// IPv4 address to listen on
+    pub listen: Ipv4Addr,
+    /// Port to run the service on
+    pub port: u16,
+    /// Directory to use for storing JSON DB state
+    pub data_dir: PathBuf,
+    /// Relative path to a valid private key in PEM format
+    pub private_key_path: PathBuf,
+    /// Activitypub related configuration for the relay
+    pub activity_pub: ActivityPubConfig,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            listen: Ipv4Addr::new(0, 0, 0, 0),
+            port: 4242,
+            data_dir: PathBuf::from("."),
+            private_key_path: PathBuf::from("private-key.pem"),
```

Comment:
do we need a default for the key? it's always required right?

Reply:
The defaults are here for being able to write out the initial config file if there isn't one already. The alternative is to just include an example one I suppose?

Reply:
yeah i assumed the resources/config.yml was the example one :)